src/Chainweb/Pact/Backend/ChainwebPactDb.hs

```diff
@@ -801,8 +801,9 @@ initSchema = do
 
 getEndingTxId :: BlockHeight -> BlockHandler SQLiteEnv TxId
 getEndingTxId bh = callDb "getEndingTxId" $ \db -> do
-    if bh == 0
-      then return 0
+    if bh == 0 -- FIXME: what if genesisHeight > 0 for the chain?
+      then error "Chainweb.Pact.Backend.ChainwebPactDb.getEndingTxId: called with block height 0"
```

Comment:
I turned this into an error, to figure out if the code relies on it. If it does, it now would be a failure, because genesis height isn't necessarily 0.

Actually, we always had explicit `isGenesisBlock` and `isGenesisHash` predicates, because we were anticipating new chains. But, we didn't enforce their use throughout the code base.

We now have the problem that within pact service backend we don't have access to the chainweb version and the chain id, so there's no way to figure out the genesis height.